components/layout_2020/positioned.rs

```diff
@@ -492,7 +492,7 @@ impl HoistedAbsolutelyPositionedBox {
             end: box_offset.inline_end,
         };
         let inline_alignment = match inline_box_offsets.either_specified() {
-            true => AlignFlags::START | AlignFlags::SAFE,
+            true => style.clone_justify_self().0 .0,
```

Comment:
I would keep the ` | AlignFlags::SAFE`, and add it to the block alignment instead. See this in Blink or Firefox nightly:

```html
<!DOCTYPE html>
<div style="position: relative; width: 50px; height: 50px; border: 3px solid">
  <div style="position: absolute; place-self: end; background: cyan; height: 100px; width: 100px; inset: 0"></div>
</div>
```

Reply:
Mm, no, actually that would be wrong too because we need to be able to opt-in into unsafe alignment.
It's just that https://github.com/servo/servo/blob/91026444701cfd68658beb21fbf446f6ed8723e6/components/layout_2020/positioned.rs#L922

should rather check for `self.alignment.flags() != AlignFlags::UNSAFE`.

But I guess that can be left for a follow-up.

Reply:
let me know if this should be done!

Reply:
Filed #34240